### ISSUE TITLE
A warning isn't always displayed when the site_dir is within the docs_dir

### ISSUE THREAD

**Issue by aiboyles** [mkdocs/mkdocs#972]
When I perform a `mkdocs serve` with the mkdocs theme, the dropdown menus in the top nav bar all work.

When I perform a `mkdocs build` and then open the index.html, the dropdown menus no longer work (same thing with `gh-deploy`). 

I suspect this is because the generated script links in the <head> include files in the `/site/site` directory, which does not exist:

```
<link href="./css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="./css/font-awesome-4.5.0.css" rel="stylesheet">
        <link href="./css/base.css" rel="stylesheet">
        <link rel="stylesheet" href="./css/highlight.css">
        <link href="./site/css/base.css" rel="stylesheet">
        <link href="./site/css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="./site/css/font-awesome-4.5.0.css" rel="stylesheet">
        <link href="./site/css/highlight.css" rel="stylesheet">
        <link href="./site/site/css/base.css" rel="stylesheet">
        <link href="./site/site/css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="./site/site/css/font-awesome-4.5.0.css" rel="stylesheet">
        <link href="./site/site/css/highlight.css" rel="stylesheet">
```

My guess is that those last four links are breaking the page, because there are errors when the page tries to find those files.

Is there some way to fix this, or prevent the build from adding those erroneous links?

**Comment by d0ugal** [comment]
Are you building your documentation to the same directory that contains the Markdown? It looks like you are seeing nested builds. You should be seeing a warning about this if you are.

What version of MkDocs are you using? Is the project public for us to take a look?

**Comment by aiboyles** [comment]
I am using version 0.15.3. I am building the documentation in the same directory that contains the Markdown (perhaps the problem is that my `docs_dir` is set to `./`? is there a way to fix this?). I do not see any warnings when I perform a build.

The repository is just a documentation repository (https://github.com/coralproject/docs), and I would like for the documentation to live at coralproject.github.io/docs when deployed. Currently we're using ReadTheDocs, but I like the different theme options that MkDocs offers (but which ReadTheDocs does not support).

**Comment by waylan** [comment]
The YAML config file (`mkdocs.yml`) and the `site_dir` should not be within the `docs_dir`. When using the `serve` command, the `site_dir` is ignored (a separate temp dir is auto-generated and used), so you don't see the problem there.

What is curious is that you don't get a warning. Settings validation should be raising one. I wonder if the config file being in the `docs_dir` is confusing things just enough to avoid the warning. If so, we need to improve settings validation to raise an error/warning for that as well.

**Comment by aiboyles** [comment]
That's the second really helpful response I've gotten from you this week about MkDocs, waylan. Thank you.

It looks like I'll have to restructure the repository to include a `docs_dir` and then just have the `mkdocs.yml` in the base of the repo. I guess ReadTheDocs didn't seem to have a problem with it, so that masked the issue.

**Comment by waylan** [comment]
I'm reopening this to remind us to address the issue with the warning not being raised. Perhaps we need to look at improving the documentation in this area as well.

**Comment by jimporter** [comment]
I'm actually fairly surprised this doesn't work (although it's a bit of an unusual practice).

I thought the point of `mkdocs.yml` is that it specifically listed all the files that would be included in a built version of the docs. Since the Coral project's repo [before the fix](https://github.com/coralproject/docs/blob/a03c307ab91a37e161d9141e3aa0bb04040851b4/mkdocs.yml) doesn't reference `site/` in any way, I don't think it should be included.

**Comment by d0ugal** [comment]
@jimporter MkDocs copies over all static assets - how would you include images etc. otherwise? So since the documentation on built into the source directory it is seen as a static asset that is then copied over.

**Comment by jimporter** [comment]
@d0ugal There are two ways you could handle that:

1. Require all static assets to be referenced in `mkdocs.yml` (with a period of time where MkDocs issues a deprecation warning that you'll need to start doing that). You could possibly include whole directories, e.g. `images/` or globs, e.g. `*.png`.
2. When building the documentation, build a list of all the static assets that get referenced by Markdown files and include those. (This fails to include images that are only included in static HTML, though. However, you could do this as a complement to (1)).

Automatically including everything (except Markdown source files, it seems) sounds like it would be prone to error, especially if you were using another tool in conjunction with MkDocs to produce your documentation.

However, the *most* surprising part is that the CSS files are automatically being included in the document. I thought you had to add the CSS files you wanted to `extra_css`. This part smells like a bug to me, since you might want to have CSS files that *only* apply to your static HTML, not to the MkDocs-built pages.

**Comment by d0ugal** [comment]
> This part smells like a bug to me, since you might want to have CSS files that only apply to your static HTML, not to the MkDocs-built pages.

@jimporter It might smell like a bug, but it is the documented behaviour and MkDocs has done this for a long time.

I'd suggest opening a new issue to do with how static assets are copied over to the build directory if you want to discuss it further and/or propose a change.

**Comment by waylan** [comment]
As a reminder, you do not need to list any pages, CSS files or JavaScript files in your `mkdcos.yml` file. In fact the default behavior is for MkDocs to walk the `site_dir` and populate those empty settings with everything it finds. I see being able to list them explicitly in the `mkdocs.yml` file as a bonus feature which gives the user added control over the default behavior (the order of pages in the nav, title used, etc,) .

It may be that manually populating those settings would be a workaround for this bug (as they are not auto-populated when the user manually populates them--at least that's the case for `pages`, not sure about `extra-css` and `extra-javascript`), but when a user has thousands of pages, it can be understandable why that would not be a reasonable solution.

As long as we support auto-populating those settings by walking the `docs_dir`, we cannot also support the `site_dir` being in the `docs_dir`.

**Comment by waylan** [comment]
And I just realized @jimporter was primarily talking about CSS & JavaScript, while I was mostly addressing pages. My point still stands with regard to pages. I'll address CSS and JS in the new issue.

**Comment by waylan** [comment]
I'm closing this as it was resolved in #996.
